CMakeLists.txt

```diff
@@ -262,8 +262,7 @@ endif()
 ################################################################################
 set(HPX_HIDDEN_VISIBILITY_DEFAULT ON)
 if(CMAKE_COMPILER_IS_GNUCXX)
-  if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 4.6.99
-      AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.8)
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.8)
     set(HPX_HIDDEN_VISIBILITY_DEFAULT OFF)
   endif()
```

Comment:
Doesn't this patch remove support for anything less than gcc 4.8? I think we don't need this `set(HPX_HIDDEN_VISIBILITY_DEFAULT OFF)` anymore at all.